canopen/nmt.py

```diff
@@ -93,6 +96,10 @@ def state(self) -> str:
 
     @state.setter
     def state(self, new_state: str):
+        logger.warning("Accessing NmtBase.state setter is deprecated, use set_state()")
```

Comment:
The DRY principle in python guides us not to, well, repeat ourselves. Having an API with multiple set/get methods is in my opinion not ideal and thus I wanted to propose to inform the user that the property based attr access is no longer the preferred way to do it. However, I do acknowledge that this is how this library has been designed. I won't argue if we decide that either `get` methods or property based access is supported without any warnings.

Reply:
Breaking existing code is even worse than repeating yourself.  But I'd still like to see a solution where we don't need any repetition, or at least reduce it to the absolute minimum.

Haven't thought it through, but I imagine some very basic code like this to "extract" the existing functions:

    class NmtBase:
        @state.setter
        def state(self, new_state: str):
            # ...    
        set_state = state.fset

My experience with `asyncio` is nearly nil though, so no clue what it would take to make this async-compatible then.